File: mobile/familysphere_app/lib/features/family/domain/entities/family_activity.dart
class FamilyActivity {
  final String id;
  final String type;
  final String message;
  final String? actorName;
  final DateTime createdAt;

  FamilyActivity({
    required this.id,
    required this.type,
    required this.message,
    this.actorName,
    required this.createdAt,
  });
}

File: backend/src/models/FamilyActivity.ts
import { Schema, model } from 'mongoose';

const familyActivitySchema = new Schema({
    familyId: {
        type: Schema.Types.ObjectId,
        ref: 'Family',
        required: true,
    },
    actorId: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true,
    },
    actorName: {
        type: String,
        default: '',
    },
    type: {
        type: String,
        required: true,
    },
    message: {
        type: String,
        required: true,
    },
    metadata: {
        type: Schema.Types.Mixed,
        default: {},
    },
}, {
    timestamps: true,
});

export default model('FamilyActivity', familyActivitySchema);
